src/Init/Data/String/Basic.lean

```diff
@@ -195,6 +195,9 @@ structure Iterator where
 
 def mkIterator (s : String) : Iterator :=
   ⟨s, 0⟩
+  
+instance : DecidableEq Iterator := fun ⟨s₁, i₁⟩ ⟨s₂, i₂⟩ => 
```

Comment:
Why didn't you use `deriving DecidableEq` at line 195?

Reply:
I think it doesn't matter in practice the order we compare. The `s` fields are usually equal, and the runtime uses pointer equality. If this is not the case, we can simply change the order of the `s` and `i` fields in the `Iterator` declaration instead of manually providing an instance here.

Reply:
done